internal/controller/authenticator/webhookcachefiller/webhookcachefiller_test.go

```diff
@@ -100,59 +928,159 @@ func TestController(t *testing.T) {
+				Reason:  "UnableToValidate",
+				Message: "unable to validate; see other conditions for details",
+			},
+		}, conditions)
+		require.Nil(t, res)
+		require.Nil(t, err)
+	})
+
+	t.Run("temp file failure, cannot create webhook authenticator", func(t *testing.T) {
 		brokenTempFile := func(_ string, _ string) (*os.File, error) { return nil, fmt.Errorf("some temp file error") }
-		res, err := newWebhookAuthenticator(nil, brokenTempFile, clientcmd.WriteToFile)
+		conditions := []*metav1.Condition{}
+		res, conditions, err := newWebhookAuthenticator(nil, brokenTempFile, clientcmd.WriteToFile, conditions, true)
+		require.Equal(t, []*metav1.Condition{
+			{
+				Type:    "AuthenticatorValid",
+				Status:  "False",
+				Reason:  "UnableToCreateTempFile",
+				Message: "unable to create temporary file: some temp file error",
+			},
+		}, conditions)
 		require.Nil(t, res)
 		require.EqualError(t, err, "unable to create temporary file: some temp file error")
 	})
 
-	t.Run("marshal failure", func(t *testing.T) {
+	t.Run("marshal failure, cannot create webhook authenticator", func(t *testing.T) {
 		marshalError := func(_ clientcmdapi.Config, _ string) error { return fmt.Errorf("some marshal error") }
-		res, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{}, os.CreateTemp, marshalError)
+		conditions := []*metav1.Condition{}
+		res, conditions, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{}, os.CreateTemp, marshalError, conditions, true)
+		require.Equal(t, []*metav1.Condition{
+			{
+				Type:    "AuthenticatorValid",
+				Status:  "False",
+				Reason:  "UnableToMarshallKubeconfig",
+				Message: "unable to marshal kubeconfig: some marshal error",
+			},
+		}, conditions)
 		require.Nil(t, res)
 		require.EqualError(t, err, "unable to marshal kubeconfig: some marshal error")
 	})
 
-	t.Run("invalid base64", func(t *testing.T) {
-		res, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{
-			Endpoint: "https://example.com",
+	// t.Run("load kubeconfig err, not currently tested, may not be necessary to test?")
+
+	t.Run("invalid TLS config, base64 encoding err, cannot create webhook authenticator", func(t *testing.T) {
+		conditions := []*metav1.Condition{}
+		res, conditions, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{
+			Endpoint: goodEndpoint,
 			TLS:      &auth1alpha1.TLSSpec{CertificateAuthorityData: "invalid-base64"},
-		}, os.CreateTemp, clientcmd.WriteToFile)
+		}, os.CreateTemp, clientcmd.WriteToFile, conditions, true)
+		require.Equal(t, []*metav1.Condition{
+			{
+				Type:    "AuthenticatorValid",
+				Status:  "False",
+				Reason:  "InvalidTLSConfiguration",
+				Message: "invalid TLS configuration: illegal base64 data at input byte 7",
+			},
+		}, conditions)
 		require.Nil(t, res)
+		// TODO: should this trigger the sync loop again with an error, or should this have been only
+		// status and log, indicating user must correct?
```

Comment:
Let's not commit notes to self. What did you decide here?

Reply:
Ah, let me remove.
The `validateTLSBundle` && `okSoFar` mechanism will already ensure we don't loop at this point, so this can stay as-is. This function isn't attached to the controller so doesn't have a logger available.  Its fine that it returns an error since the controller is already deciding what to do in this case.